### PR TITLE
Fix the signature string with a new customer data sent to oneclick and payment method

### PR DESCRIPTION
**Pull request by HonzaSaibic**

### CODE REVIEW

File: pycsob/client.py [hunk 1]
```diff
@@ -124,11 +124,11 @@ def payment_init(self, order_no, total_amount, return_url, description, customer
             ('payMethod', 'card'),
             ('totalAmount', total_amount),
             ('currency', currency),
-            ('customer', utils.convert_keys_to_camel_case(customer_data)),
             ('closePayment', close_payment),
             ('returnUrl', return_url),
             ('returnMethod', return_method),
             ('cart', cart),
+            ('customer', utils.convert_keys_to_camel_case(customer_data)),
```
Comment (HonzaSaibic) [inline pycsob/client.py:134]: we have to follow the order, due to how  mk_msg_for_sign works for both oneclick and payment init functions. https://github.com/csob/platebnibrana/wiki/Z%C3%A1kladn%C3%AD-metody#payment-init-operation

File: pycsob/client.py [hunk 2]
```diff
@@ -231,13 +231,13 @@ def oneclick_init(self, orig_pay_id, order_no, total_amount, customer_data, curr
...
             ('origPayId', orig_pay_id),
             ('orderNo', str(order_no)),
-            ('customer', utils.convert_keys_to_camel_case(customer_data)),
             ('dttm', utils.dttm()),
             ('totalAmount', total_amount),
             ('currency', currency),
             ('description', description),
             ('returnUrl', return_url),
             ('returnMethod', return_method),
+            ('customer', utils.convert_keys_to_camel_case(customer_data)),
```
Comment (HonzaSaibic) [inline pycsob/client.py:246]: https://github.com/csob/platebnibrana/wiki/Metody-pro-OneClick-platbu#oneclick-init-operation

File: pycsob/utils.py [hunk 3]
```diff
@@ -165,3 +167,26 @@ def convert_keys_to_camel_case(data: T) -> T:
...
+
+    customer_msg = "|".join(str_or_jsbool(customer_data[key]) for key in customer if key in customer_data.keys())
+
+    if account_data := customer_data.get("account", {}):
+        account_msg = "|".join(str_or_jsbool(account_data[key]) for key in account if key in account_data.keys())
+
+    if login_data := customer_data.get("login", {}):
+        login_msg = "|".join(str_or_jsbool(login_data[key]) for key in login if key in login_data.keys())
+
+    return "|".join(filter(None, (customer_msg, account_msg, login_msg)))
```
Comment (HonzaSaibic) [inline pycsob/utils.py]: https://github.com/csob/platebnibrana/wiki/Podpis-po%C5%BEadavku-a-ov%C4%9B%C5%99en%C3%AD-podpisu-odpov%C4%9Bdi

File: pycsob/client.py (hunk 1 above)
Comment (richtluk) [reply pycsob/client.py:134]: This is quite unusual, could you please add a comment about it into code? For both oneclick and payment init?

File: pycsob/utils.py [hunk 4]
```diff
@@ -46,6 +46,8 @@ def mk_msg_for_sign(payload):
         for one in payload['cart']:
             cart_msg.extend(one.values())
         payload['cart'] = '|'.join(map(str_or_jsbool, cart_msg))
+    if 'customer' in payload and payload['customer'] not in conf.EMPTY_VALUES:
```
Comment (richtluk) [inline pycsob/utils.py]: How about this:
```suggestion
    if payload.get('customer') not in conf.EMPTY_VALUES:
```
? :)

File: pycsob/utils.py [hunk 5]
```diff
@@ -165,3 +167,29 @@ def convert_keys_to_camel_case(data: T) -> T:
...
+
+    def get_joined_values(data: dict[str, Any], keys: list) -> str:
+        """
+        Args:
+            data: payload customer data
+            keys: list with ordered keys
+        """
+        return "|".join(str_or_jsbool(data[key]) for key in keys if key in data)
+
+    customer_keys = ["name", "email", "mobilePhone"]
```
Comment (richtluk) [inline pycsob/utils.py]: Nitpick: I can see you are mixing single quote and double quote conventions. I know it is pia without automation so I let it up to you, just pointing this out.

File: tests_pycsob/test_api.py [hunk 6]
```diff
@@ -343,3 +343,164 @@ def test_convert_keys_to_camel_case_should_log_error_for_not_string_keys(self):
...
+        customer = {
+            "account": {
+                "createdAt":"2022-01-12T12:10:37+01:00",
+            }
+        }
+        assert get_customer_data_signature_message(customer) == (
+            "2022-01-12T12:10:37+01:00"
+        )
+
+    def test_get_customer_data_signature_message_should_return_ignore_invalid_keys(self):
```
Comment (richtluk) [inline tests_pycsob/test_api.py]:
```suggestion
    def test_get_customer_data_signature_message_should_ignore_invalid_keys(self):
```
?

File: pycsob/utils.py (hunk 4 above)
Comment (HonzaSaibic) [reply pycsob/utils.py]: I followed the condition structure as above, but I could use this pattern.

File: pycsob/utils.py (hunk 5 above)
Comment (HonzaSaibic) [reply pycsob/utils.py]: You are right, i've changed it to single quotes 👍

File: pycsob/client.py (hunk 1 above)
Comment (HonzaSaibic) [reply pycsob/client.py:134]: Alright, i've added the documentation link to the functions doc string.

File: tests_pycsob/test_api.py (hunk 6 above)
Comment (HonzaSaibic) [reply tests_pycsob/test_api.py]: Changed 👍

**Review by rubickcz — APPROVED** [review]

File: pycsob/client.py (hunk 1 above)
Comment (richtluk) [reply pycsob/client.py:134]: Thanks. 🙏

File: pycsob/utils.py (hunk 4 above)
Comment (richtluk) [reply pycsob/utils.py]: Ah, I didn't see that. 👍

**Review by richtluk — APPROVED** [review]

**Pull request merged**